### PR TITLE
[RDY] Allow multiple function definitions in one :execute

### PR DESCRIPTION
**Pull request by ZyX-I**
It appeared to be easy enough to not error out, but actually allow commands after `:endfunction`.

Ref #6844.

### CODE REVIEW

**Comment by ZyX-I** [comment]
Having a bunch of failures like

```
# test/functional/eval/system_spec.lua @ 167
10:58:54,397 INFO  - # Failure message: ./test/functional/ui/screen.lua:301: Row 1 did not match.
10:58:54,397 INFO  - # Expected:
10:58:54,397 INFO  - # |* ^ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | ~ |
10:58:57,921 INFO  - # | :call system("echo") |
10:58:57,921 INFO  - # Actual:
10:58:57,921 INFO  - # |*^ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |~ |
10:58:57,921 INFO  - # |:call system("echo") |
```

in QB and travis (space difference), weird.

**Comment by ZyX-I** [comment]
Reproducible locally, most likely has something to do with dedent.

**Comment by ZyX-I** [comment]
Ah, `gsub` returns more then one value to be passed to `dedent`.

File: test/functional/viml/function_spec.lua [hunk 1]
```diff
@@ -1,29 +1,221 @@
...
+      endfunction!
+      call F2()
+      call F3()
+      call F4()]])))
+    check_func('F2', 'echo 2', 2)
+    check_func('F3', 'echo 3', 2)
+    check_func('F4', 'echo 4', 2)
   end)
 end)
+-- vim: foldmarker=▶,▲
```
Comment (justinmk) [inline test/functional/viml/function_spec.lua]: is this intentional?

File: test/helpers.lua [hunk 2]
```diff
@@ -297,12 +297,13 @@ local function dedent(str)
...
   end
+  local left_indent = (' '):rep(leave_indent or 0)
   -- create a pattern for the indent
   indent = indent:gsub('%s', '[ \t]')
   -- strip it from the first line
-  str = str:gsub('^'..indent, '')
+  str = str:gsub('^'..indent, left_indent)
   -- strip it from the remaining lines
-  str = str:gsub('[\n]'..indent, '\n')
+  str = str:gsub('[\n]'..indent, '\n' .. left_indent)
```
Comment (justinmk) [inline test/helpers.lua]: why is `left_indent` is catenated with the string?

File: test/helpers.lua [hunk 3]
```diff
@@ -297,12 +297,13 @@ local function dedent(str)
     -- no minimum common indent
     return str
   end
+  local left_indent = (' '):rep(leave_indent or 0)
```
Comment (justinmk) [inline test/helpers.lua]: With `gsub(..., 0)` nothing happens. What is the purpose of calling dedent() if the indent is not un-indented?

**Comment by justinmk** [comment]
Should there be a mention in vim_diff.txt? 

testlint issue:

```
Checking /home/travis/build/neovim/neovim/test/functional/viml/function_spec.lua 1 warning
    /home/travis/build/neovim/neovim/test/functional/viml/function_spec.lua:11:36: unused argument body
```

**Comment by bfredl** [comment]
@justinmk maybe let see if bram accepts it before adding to vim_diff.txt https://groups.google.com/forum/#!topic/vim_dev/oNMz-9TPyFQ

File: test/functional/viml/function_spec.lua (hunk 1 above)
Comment (ZyX-I) [inline test/functional/viml/function_spec.lua]: There are a number of `}}}` comments. Initially also `{{{`, but they appeared not to work. For that I made sure that foldmarkers in tests will not actually be considered fold markers by having this modeline with unicode foldmarkers I frequently use.

File: test/helpers.lua (hunk 3 above)
Comment (ZyX-I) [inline test/helpers.lua]: I added additional argument to add some indent. The purpose is that I may have six-space indent in the function and yet only three-space indent coming out of `dedent`.

**Comment by ZyX-I** [comment]
AppVeyor failure seems unrelated:

```
[  FAILED  ] 1 test, listed below:
[  FAILED  ] C:/projects/neovim/test/functional\core\job_spec.lua @ 45: jobs uses &shell and &shellcmdflag if passed a string
C:/projects/neovim/test/functional\core\job_spec.lua:52: Expected objects to be the same.
Passed in:
(table) {
  [1] = 'notification'
  [2] = 'stdout'
 *[3] = {
    [1] = 0
   *[2] = {
      [1] = 'abc' } } }
Expected:
(table) {
  [1] = 'notification'
  [2] = 'stdout'
 *[3] = {
    [1] = 0
   *[2] = {
      [1] = 'abc'
     *[2] = '' } } }
stack traceback:
	C:/projects/neovim/test/functional\core\job_spec.lua:52: in function <C:/projects/neovim/test/functional\core\job_spec.lua:45>
```

**Comment by ZyX-I** [comment]
QB failures are due to “timeout” and “cancel”, in both cases at

```
04:14:01,853 INFO  - # test/functional/api/keymap_spec.lua @ 215: get_keymap returns script numbers for global maps
```

Looks unrelated as well. So this PR may be considered ready.

**Pull request merged**